backend/src/lib/aws/elasticache.ts

```diff
@@ -0,0 +1,208 @@
+import { Sha256 } from "@aws-crypto/sha256-js";
+import {
+  CreateUserCommand,
```

Comment:
@DanielHougaard as i have guessed its, it going way too much for a single provider. 

1. We need to move this logic to dynamic secret service. This should not be in lib as lib is for more reusable code.
2. Elastic cache seems to contain lot more logic than the redis provider considers. Its better we put this as another provider. Let's keep the aws change on hold or rather do it later. Because consider somebody requesting azure for the same the same provider would now handle more logic for that one too which doesn't make sense.

Reply:
3. AWS access could be in multiple ways too like using `assumeRole` too.